starter-code/contact-app/src/app/contact-list/contact-list.component.html

```diff
@@ -1,18 +1,50 @@
-<div>
-  <label for="contact-name"></label>
-  <input type="text" [(ngModel)]="newContact.name">
-</div>
-<div>
-  <label for="contact-email"></label>
-  <input type="text" [(ngModel)]="newContact.email">
-</div>
-<div>
-  <label for="contact-phone"></label>
-  <input type="text" [(ngModel)]="newContact.phoneNumber">
-</div>
-<div>
-  <label for="contact-image"></label>
-  <input type="text" [(ngModel)]="newContact.image">
+<form (ngSubmit)="addContact(form)" #form="ngForm" novalidate>
+  <div class="field" [ngClass]="{'has-error': feedbackEnabled && nameField.errors}">
+    <label for="contact-name">Name</label>
+    <input type="text" [(ngModel)]="newContact.name" name = 'name' #nameField="ngModel" required minlength="6">
```

Comment:
You are not disabling your inputs when you send your data. In this case it doesn't matter, but remember to do it when you work with a database.